src/worker/vm/vm.ts

```diff
@@ -152,6 +152,8 @@ export async function createPluginConfigVM(
             const __methods = {
                 setupPlugin: __asyncFunctionGuard(__bindMeta('setupPlugin')),
                 teardownPlugin: __asyncFunctionGuard(__bindMeta('teardownPlugin')),
+                onEvent: __asyncFunctionGuard(__bindMeta('onEvent')),
+                onSnapshot: __asyncFunctionGuard(__bindMeta('onSnapshot')),
```

Comment:
`onEvent` conveys the read-only nature of this feature well, but maybe not that this is ran at the very end of processing, when the even has been processed by all plugins (as opposed to a raw event from the endpoint for instance). So maybe something like `onCompletedEvent`, `onFinishedEvent`, `onSavedEvent`?

Reply:
I'd really like to fight entropy here again :). If we add some modifier, then it's another thing for people to remember, and  the entire `on` namespace will get annoyingly repetitive. `onCreatedEvent`, `onCreatedAction`, etc.

People who will know that `onEvent` happens after `processEvent` because in the interface we'll have a visual representation of `processEvent` plugins that happen before `onEvent` plugins that can't be reordered are at the end.

Reply:
The UI is not an issue, but this may be unclear from an API perspective for plugin developers. But I don't have a strong opinion on this, so if you definitely see plain `onEvent` as the best option, then let's go for that.

Reply:
Let's just document this well :).

Reply:
With the UI, I was getting at the idea that if you're a plugin user, you will see from the UI that there are some plugins that happen in order and some that happen after them. This would lead to some understanding of how to write plugins. However, thinking further, indeed the users and developers won't be the same people (see rudderstack export), so scratch that reasoning. 🤷